### PR TITLE
feat(transformer): Support @Injectable() on static functions

### PR DESCRIPTION
**Pull request by jakemac53**

### CODE REVIEW

**Comment by jakemac53** [comment]
@yjbanov feel free to reassign this review, but I though it would be good for you to see that I did have to add support for @Injectable on static methods to do this. If you had another way you envisioned this working let me know.

**Comment by vsavkin** [comment]
Is this change required to support `const Binding(Foo, toFactory: factoryFoo)`?

**Comment by jakemac53** [comment]
yes

**Comment by jakemac53** [comment]
Or at least, its one way of making that work. If you have another idea let me know!

**Comment by vsavkin** [comment]
It looks good. But I don't know much about transformers, so I think @yjbanov should take a look too.

File: modules/angular2/src/reflection/reflector.ts [hunk 1]
```diff
@@ -13,22 +13,26 @@ export {SetterFn, GetterFn, MethodFn} from './types';
 export {PlatformReflectionCapabilities} from './platform_reflection_capabilities';
 
 export class Reflector {
-  _typeInfo: Map<Type, any>;
+  _injectableInfo: Map<any, any>;
```
Comment (yjbanov) [inline modules/angular2/src/reflection/reflector.ts]: Isn't the type of map values always `StringMap`?

Comment (jakemac53) [inline modules/angular2/src/reflection/reflector.ts]: not sure, but I can try it out

**Comment by yjbanov** [comment]
LGTM

**Pull request merged**